### PR TITLE
Feature/json numeric type

### PR DESCRIPTION
**Pull request by rangatce**
Added JSON_NUMERIC data type that can accept both integer and real (double) arguments.

### CODE REVIEW

**Comment by rangatce** [comment]
Build failure is unrelated to my change
*****************************************

[2019-05-24 19:15:59] [build] /opt/src/src/test/main.cpp:11:10: fatal error: catch2/catch.hpp: No such file or directory
[2019-05-24 19:15:59] [build]  #include <catch2/catch.hpp>
[2019-05-24 19:15:59] [build]           ^~~~~~~~~~~~~~~~~~

File: src/jsonrpccpp/common/specification.h [hunk 1]
```diff
@@ -36,8 +36,9 @@ namespace jsonrpc
         JSON_BOOLEAN = 2,
         JSON_INTEGER = 3,
         JSON_REAL = 4,
-        JSON_OBJECT = 5,
-        JSON_ARRAY = 6
+        JSON_NUMERIC = 5,
+        JSON_OBJECT = 6,
+        JSON_ARRAY = 7
```
Comment (cinemast) [inline src/jsonrpccpp/common/specification.h]: Please don't change the ordinals of existing types. Add `JSON_NUMERIC=7`.

Comment (rangatce) [reply src/jsonrpccpp/common/specification.h]: Review comments addressed and added JSON_NUMERIC=7.
Kindly check the updated diff.

**Comment by cinemast** [comment]
Thank you 👍

**Pull request merged**